v0.20.12-no-mira build error with make

When running these build commands:
```
git clone https://github.com/steemit/steem
cd steem
git checkout v0.20.12-no-mira
cmake -DLOW_MEMORY_NODE=ON -DCMAKE_BUILD_TYPE=Release .
make steemd
```

it gives this error:
```
[ 72%] Linking CXX static library libsteem_protocol.a
[ 72%] Built target steem_protocol
Scanning dependencies of target steem_chain
[ 72%] Building CXX object libraries/chain/CMakeFiles/steem_chain.dir/database.cpp.o
In file included from /home/witness/steem/libraries/protocol/include/steem/protocol/block_header.hpp:2:0,
                 from /home/witness/steem/libraries/protocol/include/steem/protocol/steem_operations.hpp:3,
                 from /home/witness/steem/libraries/chain/database.cpp:3:
/home/witness/steem/libraries/protocol/include/steem/protocol/required_automated_actions.hpp:21:82: note: #pragma message: TODO: "Remove ifdef when first required automated action is added"
In file included from /home/witness/steem/libraries/protocol/include/steem/protocol/block_header.hpp:3:0,
                 from /home/witness/steem/libraries/protocol/include/steem/protocol/steem_operations.hpp:3,
                 from /home/witness/steem/libraries/chain/database.cpp:3:
/home/witness/steem/libraries/protocol/include/steem/protocol/optional_automated_actions.hpp:21:82: note: #pragma message: TODO: "Remove ifdef when first optional automated action is added"
In file included from /home/witness/steem/libraries/protocol/include/steem/protocol/steem_operations.hpp:3:0,
                 from /home/witness/steem/libraries/chain/database.cpp:3:
/home/witness/steem/libraries/protocol/include/steem/protocol/block_header.hpp:11:65: note: #pragma message: TODO: "Remove when automated actions are created"
In file included from /home/witness/steem/libraries/chain/include/steem/chain/account_object.hpp:12:0,
                 from /home/witness/steem/libraries/chain/include/steem/chain/steem_objects.hpp:483,
                 from /home/witness/steem/libraries/chain/database.cpp:20:
/home/witness/steem/libraries/chain/include/steem/chain/util/manabar.hpp: In function ‘void steem::chain::util::update_manabar(const PropType&, AccountType&, bool, int64_t)’:
/home/witness/steem/libraries/chain/include/steem/chain/util/manabar.hpp:127:104: note: #pragma message: TODO: "This hardfork check should not be needed. Remove after HF21 if that is the case."
In file included from /home/witness/steem/libraries/mira/include/mira/index_converter.hpp:3:0,
                 from /home/witness/steem/libraries/chain/include/steem/chain/index.hpp:3,
                 from /home/witness/steem/libraries/chain/include/steem/chain/util/sps_processor.hpp:6,
                 from /home/witness/steem/libraries/chain/database.cpp:32:
/home/witness/steem/libraries/mira/include/mira/ordered_index.hpp: At global scope:
/home/witness/steem/libraries/mira/include/mira/ordered_index.hpp:23:35: error: ‘boost::multi_index::detail::null_augment_policy’ has not been declared
 using boost::multi_index::detail::null_augment_policy;
                                   ^
/home/witness/steem/libraries/mira/include/mira/ordered_index.hpp:44:7: error: ‘null_augment_policy’ is not a member of ‘mira::multi_index::detail’
       detail::null_augment_policy>                        type;
       ^
/home/witness/steem/libraries/mira/include/mira/ordered_index.hpp:44:7: error: ‘null_augment_policy’ is not a member of ‘mira::multi_index::detail’
/home/witness/steem/libraries/mira/include/mira/ordered_index.hpp:44:34: error: template argument 6 is invalid
       detail::null_augment_policy>                        type;
                                  ^
/home/witness/steem/libraries/chain/database.cpp: In lambda function:
/home/witness/steem/libraries/chain/database.cpp:2891:109: note: #pragma message: TODO: "Copied from witness_schedule.cpp, do we want to abstract this to a separate function?"
/home/witness/steem/libraries/chain/database.cpp: At global scope:
/home/witness/steem/libraries/chain/database.cpp:3299:74: note: #pragma message: TODO: "Remove when optional automated actions are created"
/home/witness/steem/libraries/chain/database.cpp: In member function ‘void steem::chain::database::process_optional_actions(const optional_automated_actions&)’:
/home/witness/steem/libraries/chain/database.cpp:3598:56: note: #pragma message: TODO: "Optimize expiration for reindex."
/home/witness/steem/libraries/chain/database.cpp: In lambda function:
/home/witness/steem/libraries/chain/database.cpp:3844:62: note: #pragma message: TODO: "#ifndef not needed after HF 20 is live"
/home/witness/steem/libraries/chain/database.cpp: In member function ‘int steem::chain::database::match(const steem::chain::limit_order_object&, const steem::chain::limit_order_object&, const steem::protocol::price&)’:
/home/witness/steem/libraries/chain/database.cpp:4074:84: note: #pragma message: TODO:  Remove if(), do assert unconditionally after HF20 occurs
 #pragma message( "TODO:  Remove if(), do assert unconditionally after HF20 occurs" )
                                                                                    ^
/home/witness/steem/libraries/chain/database.cpp:4117:84: note: #pragma message: TODO:  Remove if(), do assert unconditionally after HF20 occurs
 #pragma message( "TODO:  Remove if(), do assert unconditionally after HF20 occurs" )
                                                                                    ^
/home/witness/steem/libraries/chain/database.cpp:4149:84: note: #pragma message: TODO:  Remove if(), do assert unconditionally after HF20 occurs
 #pragma message( "TODO:  Remove if(), do assert unconditionally after HF20 occurs" )
                                                                                    ^
/home/witness/steem/libraries/chain/database.cpp: In member function ‘bool steem::chain::database::fill_order(const steem::chain::limit_order_object&, const steem::protocol::asset&, const steem::protocol::asset&)’:
/home/witness/steem/libraries/chain/database.cpp:4221:84: note: #pragma message: TODO:  Remove if(), do assert unconditionally after HF20 occurs
 #pragma message( "TODO:  Remove if(), do assert unconditionally after HF20 occurs" )
                                                                                    ^
libraries/chain/CMakeFiles/steem_chain.dir/build.make:62: recipe for target 'libraries/chain/CMakeFiles/steem_chain.dir/database.cpp.o' failed
make[3]: *** [libraries/chain/CMakeFiles/steem_chain.dir/database.cpp.o] Error 1
CMakeFiles/Makefile2:1641: recipe for target 'libraries/chain/CMakeFiles/steem_chain.dir/all' failed
make[2]: *** [libraries/chain/CMakeFiles/steem_chain.dir/all] Error 2
CMakeFiles/Makefile2:4873: recipe for target 'programs/steemd/CMakeFiles/steemd.dir/rule' failed
make[1]: *** [programs/steemd/CMakeFiles/steemd.dir/rule] Error 2
Makefile:1098: recipe for target 'steemd' failed
make: *** [steemd] Error 2
```

Version info:
```
Ubunto 16.04
Boost version: 1.58.0
GCC Version: 4:5.3.1-1ubuntu1
```

@Jolly-Pirate suggested upgrading to Ubunto 18.04. I had been using 16.04 since it had historically been the recomended version and is still listed as the preferred OS in README.md. If we should be on 18.04 now, this should be changed.

--- a/libraries/mira/include/mira/ordered_index.hpp
+++ b/libraries/mira/include/mira/ordered_index.hpp
@@ -20,7 +20,36 @@ namespace detail{
 
 /* no augment policy for plain ordered indices */
 
+#if BOOST_VERSION >= 105900
+
 using boost::multi_index::detail::null_augment_policy;
+
+#else
+
+struct null_augment_policy
+{
+  template<typename OrderedIndexImpl>
+  struct augmented_interface
+  {
+    typedef OrderedIndexImpl type;
+  };
+
+  template<typename OrderedIndexNodeImpl>
+  struct augmented_node
+  {
+    typedef OrderedIndexNodeImpl type;
+  };
+
+  template<typename Pointer> static void add(Pointer,Pointer){}
+  template<typename Pointer> static void remove(Pointer,Pointer){}
+  template<typename Pointer> static void copy(Pointer,Pointer){}
+  template<typename Pointer> static void rotate_left(Pointer,Pointer){}
+  template<typename Pointer> static void rotate_right(Pointer,Pointer){}
+
+  template<typename Pointer> static bool invariant(Pointer){return true;}
+};
+
+#endif
 
 } /* namespace multi_index::detail */
 

